sklearn/utils/tests/test_seq_dataset.py

```diff
@@ -40,65 +43,87 @@ def assert_csr_equal_values(current, expected):
     assert_array_equal(current.indptr, expected.indptr)
 
 
-def make_dense_dataset_32():
-    return ArrayDataset32(X32, y32, sample_weight32, seed=42)
-
-
-def make_dense_dataset_64():
+def _make_dense_dataset(float_precision):
+    if float_precision == np.float32:
+        return ArrayDataset32(X32, y32, sample_weight32, seed=42)
     return ArrayDataset64(X64, y64, sample_weight64, seed=42)
 
 
-def make_sparse_dataset_32():
-    return CSRDataset32(
-        X_csr32.data, X_csr32.indptr, X_csr32.indices, y32, sample_weight32, seed=42
-    )
-
-
-def make_sparse_dataset_64():
-    return CSRDataset64(
-        X_csr64.data, X_csr64.indptr, X_csr64.indices, y64, sample_weight64, seed=42
-    )
-
-
-@pytest.mark.parametrize(
-    "dataset_constructor",
-    [
-        make_dense_dataset_32,
-        make_dense_dataset_64,
-        make_sparse_dataset_32,
-        make_sparse_dataset_64,
-    ],
-)
-def test_seq_dataset_basic_iteration(dataset_constructor):
+def _make_sparse_dataset(csr_container, float_precision):
+    if float_precision == np.float32:
+        X, y, sample_weight, csr_dataset = X32, y32, sample_weight32, CSRDataset32
+    else:
+        X, y, sample_weight, csr_dataset = X64, y64, sample_weight64, CSRDataset64
+    X = csr_container(X)
+    return csr_dataset(X.data, X.indptr, X.indices, y, sample_weight, seed=42)
+
+
+def _make_dense_datasets():
+    return [
+        _make_dense_dataset(float_precision) for float_precision in float_precisions
+    ]
+
+
+def _make_sparse_datasets():
+    return [
+        _make_sparse_dataset(csr_container, float_precision)
+        for csr_container, float_precision in product(CSR_CONTAINERS, float_precisions)
+    ]
+
+
+def _make_fused_types_datasets():
+    dense_datasets = [
+        (
+            _make_dense_dataset(float_precisions[0]),
+            _make_dense_dataset(float_precisions[1]),
+        )
+    ]
```

Comment:
isn't it the same as `_make_dense_datasets()`

Reply:
For parameterisation I need to have list 
```
[
    (32_dense 64_dense), 
    (32_sparse_m, 64_sparse_m), 
    (32_sparse_a, 64_sparse_a)
]
```
`_make_dense_datasets()` returns [32_dense 64_dense],
It can be replaced with 
```python
dense_datasets = [tuple(_make_dense_datasets())]
```
`_make_sparse_datasets()` returns [32_sparse_m, 64_sparse_m, 32_sparse_a, 64_sparse_a]
It can be replaced with 
```python
sparse_datasets = _make_sparse_datasets()
# List of tuples (32_dataset, 64_dataset)
sparse_datasets = [(sparse_datasets[i], sparse_datasets[i + 1]) for i in range(0, len(sparse_datasets), 2)]
```

Or maybe there is better option

Reply:
OK I see I think the following will be shorter:

```python
def _make_fused_types_datasets():
    all_datasets = _make_dense_datasets()` + _make_sparse_datasets()
    # group dataset by array types to get a tuple (float32, float64)  
    return (
        all_datasets[idx:idx+2] for idx in range(0, len(all_datasets), 2)
    )
```